### PR TITLE
KAFKA-15750: KRaft support in KafkaMetricReporterExceptionHandlingTest

### PR DESCRIPTION
**Pull request by linzihao1999**
The tests in KafkaMetricReporterExceptionHandlingTest in core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala updated to support KRaft.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala [hunk 1]
```diff
@@ -50,19 +52,30 @@ class KafkaMetricReporterExceptionHandlingTest extends BaseRequestTest {
     // need a quota prop to register a "throttle-time" metrics after server startup
     val quotaProps = new Properties()
     quotaProps.put(QuotaConfigs.REQUEST_PERCENTAGE_OVERRIDE_CONFIG, "0.1")
-    adminZkClient.changeClientIdConfig("<default>", quotaProps)
+
+    if (isKRaftTest()) {
+      val admin = createAdminClient()
```
Comment (tinaselenge) [inline core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala]: Should we close this admin client in tearDown() method?

Comment (linzihao1999) [reply core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala]: I think it no need to. `admin` will close via IntegrationTestHarness.teardown().

Comment (tinaselenge) [reply core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala]: I see thanks!

Comment (ex172000) [inline core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala]: Can we make this part into `QuoramTestHarness`, just like the ZK client?

Comment (linzihao1999) [reply core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala]: That's good idea. Reusing this part seems much better.

Comment (linzihao1999) [reply core/src/test/scala/unit/kafka/server/KafkaMetricReporterExceptionHandlingTest.scala]: NP

**Comment by linzihao1999** [comment]
Hi @mimaison 
Please take a look when you have time

**Review by mimaison — APPROVED** [review]
LGTM, thanks for the PR!

**Pull request merged**